2-Draft-Accepted/RFCnnnn-PowerShell-as-Login-Shell.md

```diff
@@ -0,0 +1,71 @@
+---
+RFC: RFCnnnn
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: Console
+Comments Due: June 6, 2019
+Plan to implement: Yes
+---
+
+# Enable PowerShell as Login Shell on POSIX-based systems
+
+POSIX shells on Unix-based systems use a `-l` switch to start the shell as a login shell.
+This means the shell is not being started from an existing configured environment and
+needs to execute `/etc/profile` to populate environment variables along with other
+environment changes.
+
+## Motivation
+
+    As a PowerShell user,
+    I can use PowerShell as my login shell on Unix-based systems,
+    so that I can use PowerShell everywhere.
+
+## Specification
+
+Many existing tools expect `-l` to be accepted by a shell and to act as a login
+shell.
+
+`-l` (expanded form is `-LoadProfile`) will explicitly have pwsh load the PowerShell
+profile.
+This is effectively a no-op as if this switch is not specified, pwsh will load
+the PowerShell profile and only doesn't load it if `-noprofile` is specified.
+
+This will allow tools that expect `-l` to be accepted to work.
+There is no additional work to process `/etc/profile` when `-l` is used.
+
+For cases where you do need `/etc/profile` to be processed,
+such as using pwsh as your default shell,
+the proposal is to include a Bourne shell script that can be specified as the
+shell:
+
+```sh
+#!/bin/sh -l
+exec /usr/local/bin/pwsh "$@"
```

Comment:
we will need to handle the path correctly here as it is different in a few scenarios:

```
const linuxExePath        = "/usr/bin/pwsh";
const linuxPreviewExePath = "/usr/bin/pwsh-preview";

const snapExePath         = "/snap/bin/pwsh";
const snapPreviewExePath  = "/snap/bin/pwsh-preview";

const macOSExePath        = "/usr/local/bin/pwsh";
const macOSPreviewExePath = "/usr/local/bin/pwsh-preview";
```

Reply:
Good point.  We'd have different sh scripts for each of these and publish the correct one per package

Reply:
We could consider using env variable(s) for  the path(s)

Reply:
@iSazonov I don't believe there's anything similar on Unix based systems that we can use.  It doesn't help if we create such a thing.

Reply:
My intention was to unify the scripts.

Reply:
@iSazonov the scripts are distinct per pkg we build so we could templatize it, but doesn't seem to be worth the complexity since the number is relatively small